### ISSUE TITLE
Bug with annotation display: AnnotRegion are deleted?

### ISSUE THREAD

**Issue by mscheltienne** [mne-tools/mne-qt-browser#82]
Running the latest dev version of `mne` and `mne_qt_browser`, I got a bug with annotations. The file used in the video below has some annotations setup and when scrolling I get the following error in the console:

```
>>> Traceback (most recent call last):
  File "c:\users\mathieu\documents\git\mne-qt-browser\mne_qt_browser\_pg_figure.py", line 3518, in _xrange_changed
    self._update_annotations_xrange(xrange)
  File "c:\users\mathieu\documents\git\mne-qt-browser\mne_qt_browser\_pg_figure.py", line 3570, in _update_annotations_xrange
    self.mne.plt.addItem(region)
  File "C:\Users\Mathieu\Documents\pyvenv\mne\lib\site-packages\pyqtgraph\graphicsItems\PlotItem\PlotItem.py", line 535, in addItem
    self.vb.addItem(item, *args, **vbargs)
  File "C:\Users\Mathieu\Documents\pyvenv\mne\lib\site-packages\pyqtgraph\graphicsItems\ViewBox\ViewBox.py", line 407, in addItem
    if item.zValue() < self.zValue():
RuntimeError: wrapped C/C++ object of type AnnotRegion has been deleted
```

It occurs when I scroll by selecting the box on the overview, when I cross an annotation edge.

https://user-images.githubusercontent.com/73893616/160923886-b720b5cd-3ab1-479f-97c1-b0070add865d.mp4

Weirdly enough, if I now close the plotting window and do again in the same console `raw.plot()`, the new browser works as intended and this bug is not present. And second weird point, this bug is not present if I do the same 3 commands in an IPython console. :confused:

-----------

Reproducible example with MNE dataset:

```
import mne
fname = mne.datasets.sample.data_path() / 'MEG' / 'sample' / 'sample_audvis_filt-0-40_raw.fif'
raw = mne.io.read_raw_fif(fname, preload=True)
annotations = mne.Annotations([1, 10, 20, 30], [1, 2, 1, 2], ['a', 'b', 'a', 'b'])
raw.set_annotations(annotations)
raw.plot()
```

**Comment by mscheltienne** [comment]
Running with the current main of MNE and of the browser. This bug looks gone on my macOS laptop, although I recorded it on my Windows machine, I'll check if it's still present.

**Comment by mscheltienne** [comment]
Never mind, this one still occurs with the current main of MNE and of the browser on Windows.
@marsipu What is the parent widget defined for `AnnotRegion`? I suspect both bugs might be fixed if we set it correctly.

**Comment by marsipu** [comment]
@mscheltienne I can't reproduce the bug on Windows.
`AnnotRegion` is added to the QGraphicsScene and the method `parentWidget` return `RawViewBox`.
Moreover the region shouldn't be garbage-collected because a reference is stored inside `plot.mne.regions`.

Is this bug still appearing for you?

**Comment by mscheltienne** [comment]
@marsipu I can still reproduce it on Windows (not re-tested on macOS) using Python 3.8.10 in a regular python console. Here is the `pip freeze` of the environment, and it uses the latest main of both `mne` and `mne-qt-browser`: [requirements-82.txt](https://github.com/mne-tools/mne-qt-browser/files/9333353/requirements-82.txt)

https://user-images.githubusercontent.com/73893616/184536915-14d6fe90-4651-4d07-b1d3-b4e058c4605f.mp4

Can you give it a shot with Python 3.8.10 and an environment with the same `pip` dependencies? (you can install the dep. with `pip install -r requirements-82.txt`).

**Comment by marsipu** [comment]
@mscheltienne Thank you for your persistence, I found the problem:

Only when the plot is started with `raw.plot()` in a Python-Terminal or with a similar script with `python -i <script.py>`, the problem occurs. I suspect that this "interactive" python-mode somehow leads to a premature garbage-collection of some Qt-Items (even if they have plenty of reference). With `block=True` or in IPython this problem didn't occur.

But I think it is important for the plot to work in an interactive python console, that's why I opened #139
